### PR TITLE
CLOUDSTACK-8379:add support to marvin to enable deployed zone based on t...

### PR DESCRIPTION
**Pull request by srikanteswartalluri**
...he value provided in config file

### CODE REVIEW

**Comment by runseb** [comment]
@gauravaradhye can you review this ? @srikanteswartalluri looks like Travis failed.

File: tools/marvin/marvin/deployDataCenter.py [hunk 1]
```diff
@@ -790,8 +790,13 @@ def createZones(self, zones):
...
                                                )
-                enabled = getattr(zone, 'enabled', 'True')
-                if enabled == 'True' or enabled is None:
+
+                #get the variable from config file whether to enable zone or not
+                #If "enabled" tag is not found in config file, enable the zone otherwise
+                #act as mentioned in the config file
+                enabled = str(getattr(zone, 'enabled', None))
+                enabled = enabled if enabled else 'true'
+                if enabled.lower() == 'true':
```
Comment (DaanHoogland) [inline tools/marvin/marvin/deployDataCenter.py]: should work but why not make a str('boolean') a real bool? 
  if enabled.lower == true:
    enable = True
  else:
    enable = False
  if enable:

**Comment by DaanHoogland** [comment]
I don't see that the travis failures are related to this code but the change is not so significant that push through makes sense. Would be nice to have quick iterations on these kinds of refactorings.

**Comment by nitt10prashant** [comment]
enabled = enabled if enabled else 'true' 
 will set enable to true if  enable is false or None  which is unexpected,
 ideally  enable should be false if it is false in config file else true

**Comment by remibergsma** [comment]
@srikanteswartalluri Any update on this PR?

**Comment by sanju1010** [comment]
-1
Without changes it works fine, because if we don't specify the "enabled" attribute for the zone in the config file, it will enable the zone by default. Even if the zone has the attribute "enabled" without any value , then also we enable the zone. It anything other than "true" then only we won't enable the zone.

With @srikanteswartalluri changes this functionality will be broken because he is converting the "None" to string in "getattr" method. So the following line will always have some string:
enabled = enabled if enabled else 'true' 
and if the zone object does not have the "enabled" attribute we are not calling the enableZone which is not expected.

**Comment by srikanteswartalluri** [comment]
Folks, 
i am closing the PR.

**Pull request closed without merging**